file/random_access_file_reader.cc

```diff
@@ -359,6 +356,35 @@ IOStatus RandomAccessFileReader::MultiRead(
 
     {
       IOSTATS_CPU_TIMER_GUARD(cpu_read_nanos, clock_);
+      if (rate_limiter_priority != Env::IO_TOTAL && rate_limiter_ != nullptr) {
+        assert(fs_reqs != nullptr);
+        // TODO: ideally we should request bandwith for each
+        // `FSReadRequest::len` seperately instead of requesting
+        // the sum and calling `LoopRateLimitRequestHelper()` afterwards
```

Comment:
I am not sure the TODO describes a goal that is better than what's here already. One possible goal would be looping: `RequestToken()` followed by `MultiRead()` with as many requests as possible to consume the bytes granted by the rate limiter.

Reply:
> RequestToken() followed by MultiRead() with as many requests as possible to consume the bytes granted by the rate limiter.

Will reach out offline for more clarification.

Reply:
Fixed.